.github/workflows/cleanup.yml

```diff
@@ -176,3 +189,99 @@ jobs:
                 body: body,
               });
             }
+      - name: 'IAM Request Cleanup Comment'
```

Comment:
I might split this into two steps, one for always and outcome == success and one for failure. Just makes it easier to read in my opinion.

Reply:
Thanks for advice! Agree if the code is complicated, separating out make sense. However, I was trying to combine it as much as I could here 1, for the code is simple therefore does not sacrifice readability that much, 2, for sharing some code. Let me know if you have a strong opinion on this.

Reply:
No strong opinions, just laying out my preferences and some things I find useful, see example below.

```yaml
- name: 'IAM Request Success Cleanup Comment'
  if: |-
    always() && steps.cleanup_iam.outcome == 'success'
  uses: 'actions/github-script@98814c53be79b1d30f795b907e553d8679345975' # ratchet:actions/github-script@v6
  with:
    github-token: '${{ github.token }}'
    retries: '3'
    script: |+
      const fs = require("fs");

      const iamOutFilename = "${{ runner.temp }}/${{ env.IAM_OUT_FILENAME }}";
      const req = fs.readFileSync(iamOutFilename, { encoding: "utf8" });

      const body = `**\`Access on Demand\`** - 🟩 **\`IAM\`** request cleanup succeeded.
      <details>
      <summary>Details</summary>
      Removed bindings in the request below.
      \`\`\`
      ${req}
      \`\`\`
      </details>`;

      await github.rest.issues.createComment({
        owner: context.repo.owner,
        repo: context.repo.repo,
        issue_number: "${{ github.event.pull_request.number }}",
        body: body,
      });


- name: 'IAM Request Failure Cleanup Comment'
  if: |-
    always() && steps.cleanup_iam.outcome == 'failure'
  uses: 'actions/github-script@98814c53be79b1d30f795b907e553d8679345975' # ratchet:actions/github-script@v6
  with:
    github-token: '${{ github.token }}'
    retries: '3'
    script: |+
      const fs = require("fs");

      const iamOutFilename = "${{ runner.temp }}/${{ env.IAM_OUT_FILENAME }}";
      const iamErrFilename = "${{ runner.temp }}/${{ env.IAM_ERROR_FILENAME }}";

      const req = fs.readFileSync(iamOutFilename, { encoding: "utf8" });
      const error = fs.readFileSync(iamErrFilename, { encoding: "utf8" });

      const body = `**\`Access on Demand\`** - 🟥 **\`IAM\`** request cleanup failed.
      <details>
      <summary>Details</summary>
      Failed to cleanup IAM polices of the resources in the request below.
      \`\`\`
      ${req}
      \`\`\`
      Error:
      \`\`\`
      ${error}
      \`\`\`
      </details>`;

      await github.rest.issues.createComment({
        owner: context.repo.owner,
        repo: context.repo.repo,
        issue_number: "${{ github.event.pull_request.number }}",
        body: body,
      });
```

Reply:
Updated, added some empty lines in between lines in body for it to parse correctly.